### PR TITLE
Api provides functionality to create user account

### PR DESCRIPTION
**Pull request by lukeperrin10**
PT story:
https://www.pivotaltracker.com/story/show/177903758

**User story**
As a restaurant API
In order to give visitors access to order functionality
I would like to be able to allow visitors to create an account

**Changes proposed in this pull request**

- [X] - creation of user 
- [x]  - adds token based authorization

**Important**
User database was created incorrectly on first pass, Devise token auth was not used in creation, reverted user database, then deleted it to recreate user database with Devise token auth. 

**What I have learned working on this feature:** 

- The tools are there to create pretty much anything.... use them
- Check the details 
- To keep to scope and not plan ahead outside of feature

**No packages or Gems added for this feature** 

_note: Devise_token_auth was installed on a previous pull request._

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class User < ApplicationRecord
+  extend Devise::Models
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
+  include DeviseTokenAuth::Concerns::User
+  validates_presence_of :email, :encrypted_password
```
Comment (tochman) [inline app/models/user.rb]: Not sure if you need this

File: spec/requests/api/user_can_register_account_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        email: 'user@email.com',
+        password: 'password',
+        password_confirmation: 'password'
+
+      }
+    end
+
+    it 'is expected to respond with status 201' do
+      expect(response).to have_http_status 200
+    end
```
Comment (LucasKnudsen) [inline spec/requests/api/user_can_register_account_spec.rb]: :^ )

Comment (lukeperrin10) [reply spec/requests/api/user_can_register_account_spec.rb]: Updated to 200

File: app/models/user.rb (hunk 1 above)
Comment (lukeperrin10) [reply app/models/user.rb]: Removed now

**Review by tochman — CHANGES_REQUESTED** [review]
I miss unit specs for the user.

**Comment by lukeperrin10** [comment]
Adds unit for user registration

**Review by tochman — APPROVED** [review]
Okay. Looks good now

**Pull request merged**